Shared/Get-FileContentInformation.ps1

```diff
@@ -14,41 +14,46 @@ function Get-FileContentInformation {
     begin {
         Write-Verbose "Calling: $($MyInvocation.MyCommand)"
         $allFiles = New-Object System.Collections.Generic.List[string]
+        $scriptBlock = {
+            param($FileLocations)
+            $results = @{}
+            foreach ($fileLocation in $FileLocations) {
+                $present = (Test-Path $fileLocation)
+
+                if ($present) {
+                    $content = Get-Content $fileLocation -Raw -Encoding UTF8
+                } else {
+                    $content = $null
+                }
+
+                $obj = [PSCustomObject]@{
+                    Present  = $present
+                    FileName = ([IO.Path]::GetFileName($fileLocation))
+                    FilePath = $fileLocation
+                    Content  = $content
+                }
+
+                $results.Add($fileLocation, $obj)
+            }
+            return $results
+        }
     }
     process {
         foreach ($file in $FileLocation) {
             $allFiles.Add($file)
         }
     }
     end {
-        $params = @{
-            ComputerName           = $ComputerName
-            ScriptBlockDescription = "Getting File Content Information"
-            ArgumentList           = @(, $allFiles)
-            ScriptBlock            = {
-                param($FileLocations)
-                $results = @{}
-                foreach ($fileLocation in $FileLocations) {
-                    $present = (Test-Path $fileLocation)
-
-                    if ($present) {
-                        $content = Get-Content $fileLocation -Raw -Encoding UTF8
-                    } else {
-                        $content = $null
-                    }
-
-                    $obj = [PSCustomObject]@{
-                        Present  = $present
-                        FileName = ([IO.Path]::GetFileName($fileLocation))
-                        FilePath = $fileLocation
-                        Content  = $content
-                    }
-
-                    $results.Add($fileLocation, $obj)
-                }
-                return $results
+        if ($PSSenderInfo) {
+            return & $scriptBlock $allFiles
+        } else {
+            $params = @{
+                ComputerName           = $ComputerName
+                ScriptBlockDescription = "Getting File Content Information"
+                ArgumentList           = @(, $allFiles)
+                ScriptBlock            = $scriptBlock
             }
+            return (Invoke-ScriptBlockHandler @params)
```

Comment:
Try to get rid of `Invoke-ScriptBlockHandler`. Only do this if we can do it for all of Shared.